worker/uniter/remotestate/snapshot.go

```diff
@@ -78,6 +78,10 @@ type Snapshot struct {
 	// be performed by this unit.
 	Actions []string
 
+	// ActionChanged contains a channel for each action used to
+	// signal when remote state of an action has changed.
+	ActionChanged map[string]chan struct{}
```

Comment:
The snapshot doesn't typically contain channels - the channels are in the watcher implementation. The snapshop just represents the modelled state for the unit agent to reconcile.

Reply:
Yeah I agree I don't like it, but I'm going to need to refactor the resolver a bit to allow async operations to get fed changes in new snapshots. The alternative is adding an Action watcher that the uniter can invoke while its executing the specific action. Then we can leave the action notifications watcher alone.

Reply:
It would be good to keep the solution "clean", so I'm not adverse to some refactoring if needed